app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt

```diff
@@ -81,8 +87,36 @@ class MainViewModel(appContext: Application) : AndroidViewModel(appContext) {
     private var getFileDetailsJob = Job()
     private var syncOfflineFilesJob = Job()
 
+    lateinit var uploadFilesHelper: UploadFilesHelper
+
     private fun getContext() = getApplication<MainApplication>()
 
+    fun setCurrentFolder(folder: File?) {
+        folder?.let {
+            it.saveToSavedStateHandle(savedStateHandle)
```

Comment:
I don't really like passing the savedStateHandle as argument
I find it a bit strange to have it in another file
I'd prefer it the other way around, with a function in this viewModel that use a with on the parameter I think 🤔

Reply:
We can poll the others on this if you want